packages/shell-api/src/field-level-encryption.ts

```diff
@@ -0,0 +1,165 @@
+import { classPlatforms, hasAsyncChild, returnsPromise, ShellApiClass, shellApiClassDefault } from './decorators';
+import { Document, ReplPlatform, BinaryType } from '@mongosh/service-provider-core';
+import Collection from './collection';
+import Cursor from './cursor';
+import { DeleteResult } from './result';
+import { assertArgsDefined, assertArgsType } from './helpers';
+import { MongoshInvalidInputError } from '@mongosh/errors';
+
+/** Configuration options for using 'aws' as your KMS provider */
+declare interface awsKms {
+  aws: {
+    /** The access key used for the AWS KMS provider */
+    accessKeyId?: string;
+    /** The secret access key used for the AWS KMS provider */
+    secretAccessKey?: string;
+  };
+}
+
+/** Configuration options for using 'local' as your KMS provider */
+declare interface localKms {
+  local: {
+    /** The master key used to encrypt/decrypt data keys. A 96-byte long Buffer. */
+    key: BinaryType;
+  };
+}
+
+export interface ClientSideFieldLevelEncryptionOptions {
+  keyVaultClient?: any, /** This should be a Mongo class instance, but to avoid circular ref use any **/
+  keyVaultNamespace: string,
+  kmsProvider: awsKms | localKms,
+  schemaMap?: Document,
+  bypassAutoEncryption?: boolean;
+}
+
+@shellApiClassDefault
+@hasAsyncChild
+@classPlatforms([ ReplPlatform.CLI ] )
+export class ClientEncryption extends ShellApiClass {
+  public _mongo: any; // Mongo but any to avoid circular ref
+  public _libmongocrypt: any;
+
+  constructor(mongo: any) {
+    super();
+    this._mongo = mongo;
+    this._libmongocrypt = new mongo._serviceProvider.fle.ClientEncryption(
```

Comment:
Btw, I just picked `serviceProvider.fle` as a place to put the libmongocrypt bindings because that was easy – feel free to move it/put it behind a method/… if you like

Reply:
This works fine for me, since for now this is the only way we'll be doing FLE. In the future if we end up with different FLE backends for different platforms we can abstract some of the API behind methods.